oktoberfest/data/spectra.py

```diff
@@ -110,33 +136,37 @@ def add_matrix_from_hdf5(self, intensity_data: pd.DataFrame, fragment_type: Frag
         :param intensity_data: intensity sparse matrix
         :param fragment_type: choose predicted, raw, or mz
         """
-        # generate column names and build dataframe from sparse matrix
-        columns = self._gen_column_names(fragment_type)
-        intensity_data.columns = columns
-        self.add_columns(intensity_data)
+        # add sparse matrix of intensities to corresponding layer
+        if fragment_type.value == 1:
+            self.spectra_data.layers["pred_int"] = scipy.sparse.csr_matrix(intensity_data)
+        elif fragment_type.value == 2:
+            self.spectra_data.layers["raw_int"] = scipy.sparse.csr_matrix(intensity_data)
+        else:
+            self.spectra_data.X = scipy.sparse.csr_matrix(intensity_data)
 
-    def add_matrix(self, intensity_data: pd.Series, fragment_type: FragmentType) -> None:
+    def add_matrix(self, intensity_data, fragment_type: FragmentType, annotation=None) -> None:
         """
         Concatenate intensity df as a sparse matrix to our data.
 
         :param intensity_data: intensity numpy array to add
         :param fragment_type: choose predicted, raw, or mz
         """
-        intensity_df = intensity_data.explode()
-
-        # reshape based on the number of fragments
-        intensity_array = intensity_df.values.astype(np.float32).reshape(-1, c.VEC_LENGTH)
-
+        intensity_array = np.array(intensity_data)
         # Change zeros to epislon to keep the info of invalid values
         # change the -1 values to 0 (for better performance when converted to sparse representation)
         intensity_array[intensity_array == 0] = c.EPSILON
         intensity_array[intensity_array == -1] = 0.0
 
-        # generate column names and build dataframe from sparse matrix
-        intensity_df = pd.DataFrame.sparse.from_spmatrix(coo_matrix(intensity_array, dtype=np.float32))
-        columns = self._gen_column_names(fragment_type)
-        intensity_df.columns = columns
-        self.add_columns(intensity_df)
+        if annotation:
+            self.spectra_data.layers["pred_int"] = csr_matrix(intensity_array.shape)
```

Comment:
don't hardcode the layer! This depends on the FragmentType, can also be raw or mzs. Or anything else really. We may refactor this alltogether to use the functions anndata provides directly in the future, but for now, it has to work.

Reply:
also, you need to check if the layer already exists no matter if you have annotation or not. I.e.

```
if self.spectra_data.layers[<layer_name>] is None:
    self.spectra_data.layers[<layer_name>] = csr_matrix(shape)
```

Reply:
Possiblefor all layers now and checking if layer exists.